src/views/collections/CollectionView/CollectionView.vue

```diff
@@ -0,0 +1,151 @@
+          <div
+            class="puzzle-description"
+            style="overflow-wrap: break-word;"
+            v-dompurify-html="collection.desc"
+          />
+        </div>
+      </div>
+    </div>
+    <h2>
+      {{ $t('nav-bar:puzzles') }}
+    </h2>
+    <Gallery sm="4" md="3" v-if="puzzles">
+      <PuzzleCard
+        v-for="puzzle in puzzles"
+        :key="puzzle.puzzle.id"
+        :nid="puzzle.puzzle.id"
+        v-bind="puzzle.puzzle"
+        :cleared="puzzleCleared(puzzle.puzzle.id)"
+      />
+    </Gallery>
+    <div v-else>
+      <Preloader />
+    </div>
+    <Comments :comments="comments" :nid="collection.nid" />
+
+    <template #sidebar="{ isInSidebar }">
+      <SidebarPanel
+        :isInSidebar="isInSidebar"
+        :header="$t('collection-info:side-bar-header')"
+        headerIcon="@/assets/info.svg"
+      >
+        <template #header-icon>
+          <img src="@/assets/info.svg" />
+        </template>
+        <ul style="padding: 0; list-style-type:none" v-if="collection">
+          <li>
+            <img :src="collection.founder_picture" class="icon" />{{ collection.founder_name }}
+          </li>
+          <li>
+            <img src="@/assets/calendar.svg" class="icon" />{{ collection.created }}
+          </li>
+          <div v-if="editRights">
+            <li>  
+              <b-button
+              type="submit"
+              variant="primary"
+              class="submit-button"
+              :href="`/collections/${nid}/edit`"
+              >
+                {{ "Edit" }}
+              </b-button>
+            </li>
+          </div>
+        </ul>
+      </SidebarPanel>
+    </template>
+  </EternaPage>
+  <Preloader v-else style="margin-top: 10rem;" />
+</template>
+
+<script lang="ts">
+  import { Component, Vue, Mixins } from 'vue-property-decorator';
+  import EternaPage from '@/components/PageLayout/EternaPage.vue';
+  import TagsPanel from '@/components/Sidebar/TagsPanel.vue';
+  import PuzzleCard from '@/components/Cards/PuzzleCard.vue';
+  import CollectionCard from '@/components/Cards/QuestCard.vue';
+  import SidebarPanel from '@/components/Sidebar/SidebarPanel.vue';
+  import Preloader from '@/components/PageLayout/Preloader.vue';
+  import { PuzzleList, PuzzleItem, ClearedPuzzle, RoadmapAchievement, CollectionItem, CollectionResponse, CommentItem } from '@/types/common-types';
+  import FetchMixin from '@/mixins/FetchMixin';
+  import Comments from '@/components/PageLayout/Comments.vue';
+
+  @Component({
+    components: {
+      EternaPage,
+      TagsPanel,
+      PuzzleCard,
+      CollectionCard,
+      SidebarPanel,
+      Preloader,
+      Comments,
+    },
+  })
+  export default class CollectionView extends Mixins(FetchMixin) {
+    puzzles: PuzzleItem[] = [];
+
+    cleared: ClearedPuzzle[] = [];
+
+    collection: CollectionItem | null = null;
+
+    comments: CommentItem[] = [];
+
+    editRights: Boolean = false;
+
+    nid = this.$route.params.id;
```

Comment:
Either this should be a getter (probably the better option to avoid it ever being undefined) or it should be left undefined until filled out in fetch. Treating this as a default value is likely to cause bugs